scripts/get-helm-3

```diff
@@ -114,6 +114,10 @@ checkDesiredVersion() {
     elif [ "${HAS_WGET}" == "true" ]; then
       TAG=$(wget $latest_release_url -O - 2>&1 | grep 'href="/helm/helm/releases/tag/v3.[0-9]*.[0-9]*\"' | sed -E 's/.*\/helm\/helm\/releases\/tag\/(v[0-9\.]+)".*/\1/g' | head -1)
     fi
+    if [ "x$TAG" == "x" ]; then
```

Comment:
I tested this out and got the error, but in doing so I realized that this error doesn't tell me what went wrong. There's no way for me to guess how to fix this.

What if we capture the error and display it if there's a failure:
```bash
    if [ "${HAS_CURL}" == "true" ]; then
        {
            IFS=$'\n' read -r -d '' ERR;
            IFS=$'\n' read -r -d '' TAG;
        } < <((printf '\0%s\0' "$(curl -Ls --fail --show-error $latest_release_url | grep 'href="/helm/helm/releases/tag/v3.[0-9]*.[0-9]*"' | sed -E 's/.*\/helm\/helm\/releases\/tag\/(v[0-9\.]+)".*/\1/g' | head -1)" 1>&2) 2>&1)
    elif [ "${HAS_WGET}" == "true" ]; then
        {
            IFS=$'\n' read -r -d '' ERR;
            IFS=$'\n' read -r -d '' TAG;
        } < <((printf '\0%s\0' "$(wget $latest_release_url -O - | grep 'href="/helm/helm/releases/tag/v3.[0-9]*.[0-9]*"' | sed -E 's/.*\/helm\/helm\/releases\/tag\/(v[0-9\.]+)".*/\1/g' | head -1)" 1>&2) 2>&1)
    fi
    if [ "$TAG" == "" ]; then
      printf "Could not retrieve the latest release tag information from %s: %s.\n" "${latest_release_url}" "${ERR}"
      exit 1
    fi
```

Reply:
Thanks! Your implementation seemed too complicated to me so I have implemented it simpler, hope you like it.

Reply:
I also switched to using Github JSON API to avoid parsing html with grep+sed, json response would be probably more stable.